vsintegration/src/FSharp.ProjectSystem.FSharp/ProjectSystem.fsproj

```diff
@@ -42,25 +41,50 @@
   </ItemGroup>
 
   <ItemGroup>
-    <AssemblyLevelAttribute Include="Microsoft.VisualStudio.Shell.ProvideBindingRedirectionAttribute">
+    <AssemblyAttribute Include="Microsoft.VisualStudio.Shell.ProvideBindingRedirectionAttribute">
       <AssemblyName>FSharp.Core</AssemblyName>
       <OldVersionLowerBound>2.0.0.0</OldVersionLowerBound>
       <OldVersionUpperBound>$(FSCoreVersion)</OldVersionUpperBound>
       <NewVersion>$(FSCoreVersion)</NewVersion>
-      <CodeBase>$PackageFolder$\FSharp.Core.dll</CodeBase>
-    </AssemblyLevelAttribute>
-    <AssemblyLevelAttribute Include="Microsoft.VisualStudio.Shell.ProvideBindingRedirectionAttribute">
+    </AssemblyAttribute>
+    <AssemblyAttribute Include="Microsoft.VisualStudio.Shell.ProvideBindingRedirectionAttribute">
       <AssemblyName>FSharp.ProjectSystem.FSharp</AssemblyName>
       <OldVersionLowerBound>15.0.0.0</OldVersionLowerBound>
       <OldVersionUpperBound>$(VSAssemblyVersion)</OldVersionUpperBound>
       <NewVersion>$(VSAssemblyVersion)</NewVersion>
-    </AssemblyLevelAttribute>
-    <AssemblyLevelAttribute Include="Microsoft.VisualStudio.Shell.ProvideBindingRedirectionAttribute">
+    </AssemblyAttribute>
+    <AssemblyAttribute Include="Microsoft.VisualStudio.Shell.ProvideBindingRedirectionAttribute">
       <AssemblyName>FSharp.ProjectSystem.PropertyPages</AssemblyName>
       <OldVersionLowerBound>15.0.0.0</OldVersionLowerBound>
       <OldVersionUpperBound>$(VSAssemblyVersion)</OldVersionUpperBound>
       <NewVersion>$(VSAssemblyVersion)</NewVersion>
-    </AssemblyLevelAttribute>
+    </AssemblyAttribute>
+
+    <AssemblyAttribute Include="Microsoft.VisualStudio.Shell.ProvideCodeBaseAttribute">
+      <AssemblyName>FSharp.Core</AssemblyName>
+      <Version>$(FSCoreVersion)</Version>
+      <CodeBase>$PackageFolder$\FSharp.Core.dll</CodeBase>
+    </AssemblyAttribute>
+    <AssemblyAttribute Include="Microsoft.VisualStudio.Shell.ProvideCodeBaseAttribute">
+      <AssemblyName>FSharp.ProjectSystem.FSharp</AssemblyName>
+      <Version>$(VSAssemblyVersion)</Version>
+      <CodeBase>$PackageFolder$\FSharp.ProjectSystem.FSharp.dll</CodeBase>
+    </AssemblyAttribute>
+    <AssemblyAttribute Include="Microsoft.VisualStudio.Shell.ProvideCodeBaseAttribute">
+      <AssemblyName>FSharp.Compiler.Private</AssemblyName>
+      <Version>$(FSProductVersion)$(_BuildNumberSuffix)</Version>
```

Comment:
Why the difference in assembly suffix?  `$(VSAssemblyVersion)` vs `$(FSProductVersion)$(_BuildNumberSuffix)`.

Reply:
VSAssemblyVersion is 15.9.0.0
$(FSProductVersion)$(_BuildNumberSuffix) 10.2.3.0